### PR TITLE
Adding signature txn to consistency spec 

### PR DESCRIPTION
**Pull request by heidihoward**
needed for https://github.com/microsoft/CCF/pull/6124

### CODE REVIEW

File: tla/consistency/SingleNode.tla [hunk 1]
```diff
@@ -101,15 +101,23 @@ StatusCommittedResponseAction ==
             )
     /\ UNCHANGED ledgerBranches
 
+\* Append a transaction to the ledger which does not impact the state we are considering
+AppendSigTxnAction ==
```
Comment (achamayou) [inline tla/consistency/SingleNode.tla:105]:
```suggestion
AppendOtherTxnAction ==
```

File: tla/consistency/SingleNode.tla [hunk 2]
```diff
@@ -101,15 +101,23 @@ StatusCommittedResponseAction ==
...
+
 \* A CCF service with a single node will never have a view change
 \* so the log will never be rolled back and thus transaction IDs cannot be invalid
 NextSingleNodeAction ==
     \/ RwTxRequestAction
     \/ RwTxExecuteAction
     \/ RwTxResponseAction
     \/ StatusCommittedResponseAction
-
+    \/ AppendSigTxnAction
```
Comment (achamayou) [inline tla/consistency/SingleNode.tla:118]:
```suggestion
    \/ AppendOtherTxnAction
```

**Review by achamayou — APPROVED** [review]

File: tla/consistency/SingleNode.tla [hunk 3]
```diff
@@ -101,6 +101,13 @@ StatusCommittedResponseAction ==
             )
     /\ UNCHANGED ledgerBranches
 
+\* Append a transaction to the ledger which does not impact the state we are considering
+AppendNoopTxnAction ==
+    /\ \E view \in DOMAIN ledgerBranches:
+        ledgerBranches' = [ledgerBranches EXCEPT ![view] =
```
Comment (lemmy) [inline tla/consistency/SingleNode.tla:113]: Given that ledgerBranches is changed, consider commenting in more detail why this is noop, i.e., why does a transaction not impact the state and what is this state.

Comment (achamayou) [reply tla/consistency/SingleNode.tla:113]: Other Txn cover three buckets:

1. Governance transactions, which are namespaced away from application space
2. Signatures transactions, ditto
3. Other application transactions, using a different tables and/or different keys in the ledger

The last one could manifest itself if we ran multiple concurrent instances of the TVC (on separate keys of course), and tried to validate each trace individually. It doesn't sound very useful, but it becomes possible too.

File: tla/consistency/SingleNode.tla [hunk 4]
```diff
@@ -19,7 +19,7 @@ LedgerTypeOK ==
             \* Each ledger entry is tuple containing a view and tx
             \* The ledger entry index is the sequence number
             /\ ledgerBranches[view][seqnum].view \in Views
-            /\ ledgerBranches[view][seqnum].tx \in Txs
+            \* /\ ledgerBranches[view][seqnum].tx \in Txs
```
Comment (achamayou) [inline tla/consistency/SingleNode.tla:22]: Perhaps

```suggestion
            /\ "tx" \in DOMAIN ledgerBranches[view][seqnum] => ledgerBranches[view][seqnum].tx \in Txs
```

**Comment by achamayou** [comment]
Merging now to unblock TV, further improvements can be made in later PRs.

**Pull request merged**